From f1f27655fd22788efa221e392fbb571fb8ceaf05 Mon Sep 17 00:00:00 2001
From: Evgeny Mandrikov <mandrikov@gmail.com>
Date: Thu, 6 Jan 2022 17:01:05 +0100
Subject: More C++20 fixes

Partial backport of llvm-project patch:
    commit 2ccf0b76bcaf0895e04f14e3ff53c59dd96f9f0f
    Author: Evgeny Mandrikov <mandrikov@gmail.com>
    Date:   Thu Jan 6 17:01:05 2022 +0100

    Fix build failure with GCC 11 in C++20 mode

    See https://wg21.link/cwg2237

    Reviewed By: shafik, dexonsmith

    Differential Revision: https://reviews.llvm.org/D115355

Bug: chromium:819294
---
 third_party/llvm-10.0/llvm/include/llvm/CodeGen/LiveInterval.h  | 2 +-
 .../llvm-10.0/llvm/include/llvm/Support/BinaryStreamArray.h     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/third_party/llvm-10.0/llvm/include/llvm/CodeGen/LiveInterval.h b/third_party/llvm-10.0/llvm/include/llvm/CodeGen/LiveInterval.h
index 2f0b4f751..a73ccb633 100644
--- a/third_party/llvm-10.0/llvm/include/llvm/CodeGen/LiveInterval.h
+++ b/third_party/llvm-10.0/llvm/include/llvm/CodeGen/LiveInterval.h
@@ -718,7 +718,7 @@ namespace llvm {
       T *P;
 
     public:
-      SingleLinkedListIterator<T>(T *P) : P(P) {}
+      SingleLinkedListIterator(T *P) : P(P) {}
 
       SingleLinkedListIterator<T> &operator++() {
         P = P->Next;
diff --git a/third_party/llvm-10.0/llvm/include/llvm/Support/BinaryStreamArray.h b/third_party/llvm-10.0/llvm/include/llvm/Support/BinaryStreamArray.h
index 1634983d2..da18c9b2e 100644
--- a/third_party/llvm-10.0/llvm/include/llvm/Support/BinaryStreamArray.h
+++ b/third_party/llvm-10.0/llvm/include/llvm/Support/BinaryStreamArray.h
@@ -324,7 +324,7 @@ public:
   FixedStreamArrayIterator(const FixedStreamArray<T> &Array, uint32_t Index)
       : Array(Array), Index(Index) {}
 
-  FixedStreamArrayIterator<T>(const FixedStreamArrayIterator<T> &Other)
+  FixedStreamArrayIterator(const FixedStreamArrayIterator<T> &Other)
       : Array(Other.Array), Index(Other.Index) {}
   FixedStreamArrayIterator<T> &
   operator=(const FixedStreamArrayIterator<T> &Other) {
-- 
2.37.2

